From 10e3e9756898fa5f274fea3901b863f44602e611 Mon Sep 17 00:00:00 2001
From: Sven van Haastregt <sven.vanhaastregt@arm.com>
Date: Fri, 22 Apr 2022 11:51:43 +0100
Subject: [PATCH 032/172] [CI] Temporarily allow use of deprecated functions

LLVM commit c99424f76560 ("[IR] Deprecate
Type::getPointerElementType() (NFC)", 2022-04-20) caused out-of-tree
builds to fail (as these use `-Werror`).  Replacing all
`Type::getPointerElementType` calls is a longer term effort so turn
deprecation errors back into warnings to resume out-of-tree builds.

(cherry picked from commit 55d1de820841e6d9d1c6ca0cd534323d69a1cbf1)
---
 .github/workflows/check-out-of-tree-build.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/check-out-of-tree-build.yml b/.github/workflows/check-out-of-tree-build.yml
index f7a8bcf0..186de2c1 100644
--- a/.github/workflows/check-out-of-tree-build.yml
+++ b/.github/workflows/check-out-of-tree-build.yml
@@ -75,7 +75,7 @@ jobs:
           mkdir build && cd build
           cmake ${{ github.workspace }}/SPIRV-LLVM-Translator \
             -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
-            -DCMAKE_CXX_FLAGS="-Werror" \
+            -DCMAKE_CXX_FLAGS="-Werror -Wno-error=deprecated-declarations" \
             -DLLVM_INCLUDE_TESTS=ON \
             -DLLVM_EXTERNAL_LIT="/usr/lib/llvm-${{ env.LLVM_VERSION }}/build/utils/lit/lit.py" \
             -DLLVM_EXTERNAL_PROJECTS="SPIRV-Headers" \
-- 
2.20.1

